recipes-kernel/linux/linux-adi/0001-sc59x-fastautomdix-support.patch

```diff
@@ -13,85 +13,71 @@ index 25acefc408c2..9c1ac91c47b5 100644
-+			led_gpio = <34>; /* PC_02 / LED8 / LED2_KIT */
- 		};
--*/
- 
- 		core1-rproc@0x3108c000 {
- 			compatible = "adi,remoteproc";
+    possible trace noise/interference?
 diff --git a/drivers/net/phy/dp83867.c b/drivers/net/phy/dp83867.c
-index 3aa1ae211f5f..d03db6405499 100644
+index 3aa1ae211f5f..d8a24bfb9928 100644
 --- a/drivers/net/phy/dp83867.c
 +++ b/drivers/net/phy/dp83867.c
 @@ -23,6 +23,8 @@
  #define MII_DP83867_ISR		0x13
  #define DP83867_CTRL		0x1f
  #define DP83867_CFG3		0x1e
-+#define DP83867_CFG3_FASTLINK   (BIT(15))
-+#define DP83867_CFG3_FAST_AUT_MDIX  (BIT(8))
++#define DP83867_CFG3_FASTLINK	(BIT(15))
++#define DP83867_CFG3_FAST_AUT_MDIX	(BIT(8))
  
  /* Extended Registers */
  #define DP83867_CFG4            0x0031
 @@ -424,6 +426,10 @@ static int dp83867_config_init(struct phy_device *phydev)
  		phy_write(phydev, DP83867_CFG3, val);
  #if (!defined(CONFIG_ARCH_SC58X)) && (!defined(CONFIG_ARCH_SC57X))
  	}
-+         /*Fast Auto MDIX is enabled for faster link up*/
-+        val = phy_read(phydev, DP83867_CFG3);
-+        val |= (DP83867_CFG3_FASTLINK | DP83867_CFG3_FAST_AUT_MDIX);
-+        phy_write(phydev, DP83867_CFG3, val);
++	/* Fast Auto MDIX is enabled for faster link up*/
++	val = phy_read(phydev, DP83867_CFG3);
++	val |= (DP83867_CFG3_FASTLINK | DP83867_CFG3_FAST_AUT_MDIX);
++	phy_write(phydev, DP83867_CFG3, val);
  #endif
  	if (dp83867->port_mirroring != DP83867_PORT_MIRROING_KEEP)
  		dp83867_config_port_mirroring(phydev);
 diff --git a/drivers/net/phy/phy.c b/drivers/net/phy/phy.c
-index 105d389b58e7..0598fec97ca7 100644
+index 105d389b58e7..77db6a3f1dde 100644
 --- a/drivers/net/phy/phy.c
 +++ b/drivers/net/phy/phy.c
 @@ -951,8 +951,9 @@ void phy_state_machine(struct work_struct *work)
  	 * called from phy_disconnect() synchronously.
  	 */
  	mutex_lock(&phydev->lock);
-+	/* Set the polling interval to 20msec*/
++	/* set the polling interval to 20msec*/
  	if (phy_polling_mode(phydev) && phy_is_started(phydev))
 -		phy_queue_state_machine(phydev, PHY_STATE_TIME);
 +		phy_queue_state_machine(phydev, PHY_STATE_TIME/50);
  	mutex_unlock(&phydev->lock);
  }
  
 diff --git a/drivers/net/phy/phy_device.c b/drivers/net/phy/phy_device.c
-index adb66a2fae18..76e64f1b5ca2 100644
+index adb66a2fae18..dc9dcfd3af42 100644
 --- a/drivers/net/phy/phy_device.c
 +++ b/drivers/net/phy/phy_device.c
 @@ -596,14 +596,14 @@ struct phy_device *phy_device_create(struct mii_bus *bus, int addr, int phy_id,
  	mdiodev->device_free = phy_mdio_device_free;
  	mdiodev->device_remove = phy_mdio_device_remove;
  
 -	dev->speed = 0;
-+	dev->speed = SPEED_100; /*Changed to 100mbps from default 0*/
++	dev->speed = SPEED_100; /*Disabling Autonegotiation by default*/
  	dev->duplex = -1;
  	dev->pause = 0;
  	dev->asym_pause = 0;
  	dev->link = 0;
  	dev->interface = PHY_INTERFACE_MODE_GMII;
  
 -	dev->autoneg = AUTONEG_ENABLE;
-+	dev->autoneg = AUTONEG_DISABLE; /*Disabling Autonegotiation by default*/
++	dev->autoneg = AUTONEG_DISABLE;
```

Comment:
Don't set anything as default. Let the previous default be as it is.

Reply:
Removed default Auto negotiation setting to default. We can handle that change in script.